scripts/monitor_captcha_safety.py

```
@@ -22,188 +22,219 @@ def __init__(self, log_file: str = "logs/captcha_safety.log"):
         self.lock = threading.Lock()
         self.is_monitoring = False
 
-    def log_captcha_attempt(self, success: bool, captcha_type: str, cost: float = 0.0,
-                          response_time: float = 0.0, error: str = None):
+    def log_captcha_attempt(
+        self,
+        success: bool,
+        captcha_type: str,
+        cost: float = 0.0,
+        response_time: float = 0.0,
+        error: str = None,
```

Ruff (0.15.1)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)